presto-array/src/test/java/com/facebook/presto/array/TestBlockBigArray.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.array;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.BlockBuilderStatus;
+import com.facebook.presto.spi.block.IntArrayBlockBuilder;
+import org.openjdk.jol.info.ClassLayout;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+public class TestBlockBigArray
+{
+    @Test
+    public void testSizeOverlapping()
+    {
+        // Create a block with some values
+        int entries = 100;
+        BlockBuilder blockBuilder = new IntArrayBlockBuilder(new BlockBuilderStatus(), entries);
+        for (int i = 0; i < entries; i++) {
+            blockBuilder.writeInt(i);
+        }
+        Block block = blockBuilder.build();
+
+        // Verify we do not over count
+        int blocks = 6789;
```

Comment:
You can use a random integer instead of a hardcoded `arraySize` value (this also applies to the `entries` constant above).
 
```
int arraySize = ThreadLocalRandom.current().nextInt(1000);
BlockBigArray blockBigArray = new BlockBigArray();
blockBigArray.ensureCapacity(arraySize);
...
```